Doc/library/argparse.rst

```diff
@@ -1723,7 +1723,7 @@ Sub-commands
    :class:`ArgumentParser` supports the creation of such sub-commands with the
    :meth:`add_subparsers` method.  The :meth:`add_subparsers` method is normally
    called with no arguments and returns a special action object.  This object
-   has a single method, :meth:`~ArgumentParser.add_parser`, which takes a
+   has a single method, :meth:`!add_parser`, which takes a
```

Comment:
```suggestion
   has a single method, :meth:`add_parser`, which takes a
```

Unsilence this, since it is a legitimate warning of missing documentation

Reply:
*(I saw this PR on the discord and it is now the last part of my 'day of procrastination' -- sorry for my absence of late, hopefully will be more around towards the end of the month)*

As an alternate, given that nearly everything in ``argparse.rst`` is fixed, we could add an entry to ``nitpick_ignore`` in ``conf.py`` with a detailed explanation, e.g.:

```python
nitpick_ignore = [
    # do not error nit-picky mode builds when ArgumentParser.add_parser cannot
    # be resolved, as the method is currently undocumented. For context, see
    # https://github.com/python/cpython/pull/103289.
    ('py:meth', 'ArgumentParser.add_parser'),
]
```

This would require reverting the above (and the usage on line 1792 below) to ``:meth:`~ArgumentParser.add_parser```, and would have the same benefits that once `.add_parser()` is documented, everything would work as if by magic.

[`nitpick_ignore` reference docs](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-nitpick_ignore)

A

Reply:
Good to see you back, and thanks for the input and ideas.

It sounds like that approach could definitely make sense particularly in cases where the existing object has the correct target (or can be easily corrected to such), but that target is just not yet documented and doing so is deferred to a followup PR. 

In this case, `ArgumentParser.add_parser` was actually a one-off typo (the other instances just feature a bare method, which this harmonizes them to), as `add_parser` is not a method of `ArgumentParser` at all, but rather a separate subparsers object (of class `_SubParsersAction`) returned by calling `ArgumentParser.add_subparsers()`.

In order to implement this, we could fix the existing references to point to the actual target, like so:

```suggestion
   has a single method, :meth:`~_SubParsersAction.add_parser`, which takes a
```

and add it to `nitpick_ignore` as you suggest above; then, when we fix it just remove it from `.nitignore`, though that does require a bit more effort/complexity overall than just keeping the status quo for this PR and just fixing it in the followup, and does mean an implicit commitment to implementing the specific solution I propose above (documenting it as `.. method:: _SubParsersAction.add_parser`).

Up to @hugovk on whether he wants to implement that approach here vs. just reverting the silencing for now.